.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+     - v**
+  pull_request:
+    branches: [ master ]
+     
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+      
+    - name: Log into registry ${{ env.REGISTRY }}
+      if: github.event_name == 'push'
+      uses: docker/login-action@v2.1.0
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Docker Setup QEMU
+      uses: docker/setup-qemu-action@v2.1.0
+      
+    - name: Docker Setup Buildx
+      uses: docker/setup-buildx-action@v2.5.0
+      with:
+        version: latest
+        install: true
+        
+    - name: Docker Metadata action
+      id: meta
+      if: github.event_name == 'push'
+      uses: docker/metadata-action@v4.3.0
+      with:
+        images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+        flavor: |
+          latest=${{ github.event_name == 'push' }}
+        tags: ${{ github.ref_name }}
+        
+    - name: Build and push Docker images
+      uses: docker/build-push-action@v4.0.0
+      with:
+        context: .
+        file: Dockerfile
+        platforms: linux/arm/v7,linux/arm64,linux/amd64
+        push: ${{ github.event_name == 'push' }}
+        tags: ${{ steps.meta.outputs.tags }}
+        labels: ${{ steps.meta.outputs.labels }}
+    
+    - name: Create release directory
+      run: mkdir release
+      
+    - name: Zip Release
+      if: github.event_name == 'push'
+      uses: TheDoctor0/zip-release@0.7.1
+      with:
+        type: 'zip'
+        filename: release/ankerctl.zip
```

Comment:
I'd strongly prefer we use a descriptive filename. We're going to end up with a hundred files all called `ankerctl.(zip|tar.gz`.

I'm sure there's an environment var that roughly correlates to something like `git describe`?

Reply:
you're thinking something like `ankerctl-v1.0.0.(zip|tar.gz)`? I'm pretty sure the tag that triggered the build can be pulled in here.

Reply:
That would be great, yeah :)

I'm marking this as "unresolved", since it's not yet implemented.